### PR TITLE
[Merged by Bors] - feat: generalize universes for connected categories

### PR DESCRIPTION
**Pull request by TwoFX**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by joelriou** [comment]
I had also encountered this annoying universe limitation, so that this PR is very welcome, and it looks good to me!

**Comment by joelriou** [comment]
maintainer merge

File: Mathlib/Logic/Function/Basic.lean [hunk 1]
```diff
@@ -434,6 +434,10 @@ theorem invFun_eq (h : ∃ a, f a = b) : f (invFun f b) = b :=
   by simp only [invFun, dif_pos h, h.choose_spec]
 #align function.inv_fun_eq Function.invFun_eq
 
+theorem apply_invFun_apply {α : Type u₁} {β : Type u₂} {f : α → β} {a : α} :
```
Comment (ocfnash) [inline Mathlib/Logic/Function/Basic.lean:437]: Can this be simp?
```suggestion
@[simp] theorem apply_invFun_apply {α : Type u₁} {β : Type u₂} {f : α → β} {a : α} :
```

**Comment by ocfnash** [comment]
Thanks!

bors d+

File: Mathlib/Logic/Function/Basic.lean (hunk 1 above)
Comment (TwoFX) [reply Mathlib/Logic/Function/Basic.lean:437]: Unfortunately it cannot, because the head symbol of the LHS is a variable.

**Pull request closed without merging**